### PR TITLE
Apply "readability-make-member-function-const" clang-tidy check

### PR DESCRIPTION
**Pull request by florian-kuebler**
Also mark some of the functions as nodiscard.

Test: Compile

### CODE REVIEW

**Review by dpallotti — APPROVED** [review]

File: src/CaptureClient/include/CaptureClient/GpuQueueSubmissionProcessor.h [hunk 1]
```diff
@@ -88,7 +88,7 @@ class GpuQueueSubmissionProcessor {
       const std::optional<orbit_grpc_protos::GpuCommandBuffer>& first_command_buffer,
       uint64_t timeline_hash,
       const std::function<uint64_t(std::string_view str)>&
-          get_string_hash_and_send_to_listener_if_necessary);
+          get_string_hash_and_send_to_listener_if_necessary) const;
```
[the remark this answers is not in the document]
Comment (florian-kuebler) [reply src/CaptureClient/include/CaptureClient/GpuQueueSubmissionProcessor.h:91]: I agree on your reasoning. I'd rather see it similar to nodiscard, in the sense, that const should be just the default. So you would not reason about why is this marked as const, but rather, oh this function is non-const, so it modifies state of the class.

[the remark this answers is not in the document]
Comment (dpallotti) [reply src/CaptureClient/include/CaptureClient/GpuQueueSubmissionProcessor.h:91]: Interesting view. On first thought, I would say that the difference is that, if a function returns something, you probably don't want to discard it, hence nodiscard should be the default. But the point of a business-logic class is probably going to be to keep and modify some internal state, so having const as a default doesn't make much sense (or as I said it before, constness just doesn't matter for those types).

[the remark this answers is not in the document]
Comment (florian-kuebler) [reply src/CaptureClient/include/CaptureClient/GpuQueueSubmissionProcessor.h:91]: Yeah, the comparison was not 100% fair. I was more thinking of the notion to the compiler. Having a function const (business or data class), allows the compiler to do nice optimisations. So in that sense, it should be the default (or rather no need to have it at all, but thats a different story).

Anyways. Whats your take on this? Shall I merge it, or disable to the check?

[the remark this answers is not in the document]
Comment (florian-kuebler) [reply src/CaptureClient/include/CaptureClient/GpuQueueSubmissionProcessor.h:91]: >Yeah, as I said, I don't mind the change, it was just a thought for discussion. I would also keep the check as it's useful in many cases. I would just not make it too harsh, like prevent compilation or merges, but I guess this wasn't the idea anyway.

Exactly, thats the idea. Currently all clang-tidy warnings are just warnings. At some point we might handpick some checks to make them errors. But that is not at the horizon.

**Pull request merged**